shapeshift/web
# fix(defi): select feeAssetBalance by granular AccountId

## Description

This PR fixes the `feeAssetBalance` to use a granular portfolio balance by AccountId, as opposed to selecting the aggregation over all accounts currently.

Accidentally brings multi account ish support over the Idle DeFi provider, although this only passes `AccountId` around for consistency and does not bring account bip44Params support yet, TBD in https://github.com/shapeshift/lib/issues/943

## Notice




## Pull Request Type


## Issue (if applicable)



closes https://github.com/shapeshift/web/issues/3209

## Risk



- Multi-account could be borked over the DeFi section, test accordingly

## Testing



- Have an account, either 0 or another without enough ETH for gas, but either enough dust to deposit, staked amount to withdraw, or rewards to claim
- Have some ETH balance over your other accounts to total to a significant amount to cover gas
- Ensure when depositing/withdrawing/claiming for your account without enough ETH for gas, the "Not enough ETH to cover gas" message is shown and the sign and broadcast button is disabled
- Ensure when depositing/withdrawing/claiming for another account with enough ETH for gas, it is possible to continue with the flow

### Engineering



☝🏽 

### Operations



☝🏽

## Screenshots (if applicable)

<img width="524" alt="image" src="https://user-images.githubusercontent.com/17035424/199617088-4aad5eea-4bb4-49a4-808b-a6bd01bc04cf.png">
<img width="523" alt="image" src="https://user-images.githubusercontent.com/17035424/199619077-8bb4b5d6-a999-4fd9-82ad-a6d343aa80ed.png">

<img width="524" alt="image" src="https://user-images.githubusercontent.com/17035424/199619108-642e8349-3069-4c71-9a4e-00febf4567e4.png">
<img width="529" alt="image" src="https://user-images.githubusercontent.com/17035424/199619136-8090d8d5-1c4a-458b-9f05-4b503fe7ebcd.png">

<img width="524" alt="image" src="https://user-images.githubusercontent.com/17035424/199619167-0c59b9c1-eb6c-4ddf-b153-0352c1c36240.png">
<img width="533" alt="image" src="https://user-images.githubusercontent.com/17035424/199619543-23a33cd2-6417-4731-9805-0fd543c2eeb5.png">
